### PR TITLE
feat: creating POST resource for push tx endpoint

### PR DESCRIPTION
**Pull request by pedroferreira1**
GET resource was kept for compatibility but will be deprecated

### CODE REVIEW

File: tests/resources/transaction/test_pushtx.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def test_get(self):
         self.assertFalse(data['success'])
 
         # force
-        response = yield self.web.get('push_tx', {b'hex_tx': bytes(tx.get_struct().hex(), 'utf-8'), b'force': b'true'})
+        response = yield self.web.post('push_tx', {'hex_tx': tx.get_struct().hex(), 'force': True})
```
Comment (msbrogli) [inline tests/resources/transaction/test_pushtx.py:88]: Shouldn't we keep the GET test until the support for GET is removed?

**Review by msbrogli — CHANGES_REQUESTED** [review]

**Review by obiyankenobi — APPROVED** [review]

File: tests/resources/transaction/test_pushtx.py (hunk 1 above)
Comment (pedroferreira1) [reply tests/resources/transaction/test_pushtx.py]: @msbrogli as discussed in privately, I've done a refactor in the tests to have all tests for the GET and the POST. Can you see if it's good?

File: tests/resources/transaction/test_pushtx.py [hunk 2]
```diff
@@ -56,7 +56,14 @@ def test_get(self):
         tx.parents = self.manager.get_new_tx_parents(tx.timestamp)
         tx.resolve()
 
-        response = yield self.web.get('push_tx', {b'hex_tx': bytes(tx.get_struct().hex(), 'utf-8')})
+        push_tx_fn = self.web.post if is_post else self.web.get
+        hex_param = 'hex_tx' if is_post else b'hex_tx'
```
Comment (msbrogli) [inline tests/resources/transaction/test_pushtx.py:60]: Why don't we call a function that converts all keys and values to binary if needed?

File: tests/resources/transaction/test_pushtx.py [hunk 3]
```diff
@@ -56,7 +56,14 @@ def test_get(self):
...
+        force_param = 'force' if is_post else b'force'
+
+        tx_hex = tx.get_struct().hex()
+        hex_data = tx_hex if is_post else bytes(tx_hex, 'utf-8')
```
Comment (msbrogli) [inline tests/resources/transaction/test_pushtx.py:64]: Why do we have different things? Are we sending binaries in the HTTP POST's body?

**Review by msbrogli — APPROVED** [review]

File: tests/resources/transaction/test_pushtx.py (hunk 3 above)
Comment (pedroferreira1) [reply tests/resources/transaction/test_pushtx.py:64]: No, the GET requires bytes and the POST doesn't

File: tests/resources/transaction/test_pushtx.py (hunk 2 above)
Comment (pedroferreira1) [reply tests/resources/transaction/test_pushtx.py:60]: As discussed privately we decided to keep like this

**Pull request merged**